packages/nodes-base/nodes/Schedule/ScheduleTrigger.node.ts

```diff
@@ -412,238 +411,74 @@ export class ScheduleTrigger implements INodeType {
 	};
 
 	async trigger(this: ITriggerFunctions): Promise<ITriggerResponse> {
-		const rule = this.getNodeParameter('rule', []) as IDataObject;
-		const interval = rule.interval as IDataObject[];
+		const { interval: intervals } = this.getNodeParameter('rule', []) as Rule;
 		const timezone = this.getTimezone();
-		const nodeVersion = this.getNode().typeVersion;
-		const cronJobs: CronJob[] = [];
-		const intervalArr: NodeJS.Timeout[] = [];
 		const staticData = this.getWorkflowStaticData('node') as {
-			recurrencyRules: number[];
+			recurrenceRules: number[];
 		};
-		if (!staticData.recurrencyRules) {
-			staticData.recurrencyRules = [];
+		if (!staticData.recurrenceRules) {
+			staticData.recurrenceRules = [];
 		}
-		const fallbackToZero = addFallbackValue(nodeVersion >= 1.2, '0');
```

Comment:
This was introduced to fix a behavior for schedule nodes version 1.2 and above.
For version 1.0 and 1.1 if you set `triggerAtMinute` to an empty string it will run every minute, as opposed to every hour/day/week, etc. 

This was fixed here: https://github.com/n8n-io/n8n/pull/9146
The fix added the version 1.2 and only applied to that and following versions. I'm assuming this was done so that it would not break users that rely on the buggy behavior. 

With this refactor we apply the "fixed" behavior to all versions now. Which is technically a breaking change.

@Joffcom sorry for looping you in here out of the blue, but maybe you have some insight if we can assume that this will break some workflows in the wild or if we can just "fix" it for all versions?

cc @michael-radency

Reply:
@despairblue correct, changes that could be breaking are usually scoped under version, if it's not bugfix
we need to update `packages/cli/BREAKING-CHANGES.md` in this case

Reply:
I'm gonna try to add the check back, and also add tests to prevent this from breaking in the future 🤞🏽

Reply:
> For version 1.0 and 1.1 if you set triggerAtMinute to an empty string it will run every minute, as opposed to every hour/day/week, etc.

When I try this on older versions of the node, the Cron expression that's generated is `' * * * *'`, which in an invalid expression. 
@michael-radency do you recall why did we decide to support this old behavior? 
IMO, this is bug, and shouldn't have been supported. All versions of this node should fallback to `0` when the field is an empty string.

Reply:
In the new code now if a user leaves the minute empty, we generate a valid cron expression by falling back to a random minute during activation. 

I'm still not sure why created a new version in #9146. 
I believe this was a bug, and we should have fixed this in the existing version of the node. 
If user is adding invalid values in any of the fields, then we don't owe them backward compatibility, nor do we need to document this as a breaking change.

IMO the changes proposed in this PR are backward compatible for anyone using this node as intended, for everyone else when the `minute` field is left invalid, falling back to selecting a random minute during activation works fine.